src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs

```diff
@@ -91,6 +91,7 @@ public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder
             string primaryConfigFile = null, string globalConfigFile = null, string environmentConfigFile = null, bool? optional = null, bool? reloadOnChange = null) // optional injections
         {
             var json = GetMergedOcelotJson(folder, env, null, primaryConfigFile, globalConfigFile, environmentConfigFile);
+            primaryConfigFile ??= Path.Join(folder, PrimaryConfigFile); // if not specified, merge & write back to the same folder

```

Comment:
Hi Ben,
We overlooked this change, and acceptance testing has revealed the issue. The folder information should be retained, and the file must be written back to the same folder.